src/Auth0Client.ts

```diff
@@ -638,32 +638,33 @@ export default class Auth0Client {
       }
     }
 
-    try {
-      await lock.acquireLock(GET_TOKEN_SILENTLY_LOCK_KEY, 5000);
-      // Check the cache a second time, because it may have been populated
-      // by a previous call while this call was waiting to acquire the lock.
-      if (!ignoreCache) {
-        let accessToken = getAccessTokenFromCache();
-        if (accessToken) {
-          return accessToken;
-        }
-      }
+    while (true) {
+      if (await lock.acquireLock(GET_TOKEN_SILENTLY_LOCK_KEY, 5000)) {
```

Comment:
I know that `Auth0Client` is kind of a singleton class (i.e. only one instance of it is supposed to be created at any point of time), but the fact that `lock` is a variable shared by all instances make the reading of this code a bit hard as it implies some potential clashes in case multiple instances of `Auth0Client` are created.

Reply:
I am not entirely sure I understand your point here. What would you propose as a solution to improve readability?

Reply:
i'm talking about storing `lock` on the instance of `Auth0Client` rather than re-use/share it between all instances